### PR TITLE
Added Mineclone 2 compatibility

### PR DESCRIPTION
**Pull request by wuniversales**

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -1,3 +1,15 @@
+-- check for MineClone2
+local mcl = minetest.get_modpath("mcl_core")
+
+if mcl then
+    sounditem = mcl_sounds.node_sound_wood_defaults()
+    item1 = "mcl_core:water_source"
+    stairs = mcl_stairs
```
Comment (SmallJoker) [inline init.lua]: Please don't define global variables. This might cause side-effects in other mods.

File: init.lua [hunk 2]
```diff
@@ -1,3 +1,15 @@
+-- check for MineClone2
+local mcl = minetest.get_modpath("mcl_core")
+
+if mcl then
+    sounditem = mcl_sounds.node_sound_wood_defaults()
```
Comment (SmallJoker) [inline init.lua]: Please use tabs to indent.

File: mapgen.lua [hunk 3]
```diff
@@ -1,10 +1,23 @@
+-- check for MineClone2
+local mcl = minetest.get_modpath("mcl_core")
+
+if mcl then
+    item1 = "mcl_core:water_source"
```
Comment (SmallJoker) [inline mapgen.lua]: Names like `item_water_source` or `item_dirt` would improve readability a lot for future changes.

**Comment by SmallJoker** [comment]
Thank your for this contribution. Concept is fine, and I'll merge the PR after the comments were handled.

**Comment by wuniversales** [comment]
Updated, the topic of indents is not exactly how to do, I think it appears ordered.

**Comment by SmallJoker** [comment]
Alright then I'll do it myself.

2a5d21ec9a422527a2543a8ea0bb7e3fb5521948
40145eb3242a74add9b42c1d79733948ade0daaf

**Pull request closed without merging**